host/archival_test.go

```diff
@@ -23,110 +23,111 @@ package host
+		s.validateHistoryArchived(s.archivalDomainName, workflowID, runID)
+		s.validateHistoryDeleted(domainID, workflowID, runID)
 	}
-	s.True(s.historyDeletedFromPersistence(getHistoryReq))
-	s.False(s.historyArchived(getHistoryReq))
-	// update getHistoryReq to include runID thereby enabling history to be fetched from archival
-	getHistoryReq.Execution.RunId = common.StringPtr(runID)
-	s.True(s.historyArchived(getHistoryReq))
 }
 
-func (s *integrationSuite) historyDeletedFromPersistence(getHistoryReq *workflow.GetWorkflowExecutionHistoryRequest) bool {
-	for i := 0; i < 10; i++ {
-		_, err := s.engine.GetWorkflowExecutionHistory(createContext(), getHistoryReq)
-		if err != nil {
-			_, ok := err.(*workflow.EntityNotExistsError)
-			return ok
+func (s *integrationSuite) getDomainID(domain string) string {
+	domainResp, err := s.engine.DescribeDomain(createContext(), &workflow.DescribeDomainRequest{
+		Name: common.StringPtr(s.archivalDomainName),
+	})
+	s.Nil(err)
+	return domainResp.DomainInfo.GetUUID()
+}
+
+func (s *integrationSuite) validateHistoryArchived(domain, workflowID, runID string) {
+	var getHistoryResp *workflow.GetWorkflowExecutionHistoryResponse
+	var err error
+	for i := 0; i < retryLimit; i++ {
+		getHistoryResp, err = s.engine.GetWorkflowExecutionHistory(createContext(), &workflow.GetWorkflowExecutionHistoryRequest{
+			Domain: common.StringPtr(s.archivalDomainName),
+			Execution: &workflow.WorkflowExecution{
+				WorkflowId: common.StringPtr(workflowID),
+				RunId:      common.StringPtr(runID),
+			},
+		})
+		if getHistoryResp != nil && getHistoryResp.GetArchived() {
+			break
 		}
 		time.Sleep(200 * time.Millisecond)
 	}
-	return false
+	s.NoError(err)
+	s.NotNil(getHistoryResp)
+	s.True(getHistoryResp.GetArchived())
 }
 
-func (s *integrationSuite) historyArchived(getHistoryReq *workflow.GetWorkflowExecutionHistoryRequest) bool {
-	for i := 0; i < 10; i++ {
-		getHistoryResp, _ := s.engine.GetWorkflowExecutionHistory(createContext(), getHistoryReq)
-		if getHistoryResp != nil && getHistoryResp.GetArchived() {
-			return true
+func (s *integrationSuite) validateHistoryDeleted(domainID, workflowID, runID string) {
+	var err error
+	for i := 0; i < retryLimit; i++ {
+		if !s.testClusterConfig.EnableEventsV2 {
+			_, err = s.testCluster.testBase.HistoryMgr.GetWorkflowExecutionHistory(&persistence.GetWorkflowExecutionHistoryRequest{
+				DomainID: domainID,
+				Execution: workflow.WorkflowExecution{
+					WorkflowId: common.StringPtr(workflowID),
+					RunId:      common.StringPtr(runID),
+				},
+				FirstEventID: common.FirstEventID,
+				NextEventID:  common.EndEventID,
+				PageSize:     1,
+			})
+		} else {
+			var resp *persistence.GetHistoryTreeResponse
+			resp, err = s.testCluster.testBase.HistoryV2Mgr.GetHistoryTree(&persistence.GetHistoryTreeRequest{
+				TreeID: runID,
+			})
+			s.Nil(err)
+			if len(resp.Branches) == 0 && len(resp.ForkingInProgressBranches) == 0 {
+				err = &workflow.EntityNotExistsError{}
```

Comment:
This feels odd to be converting the error type. Can we also just have this method return a boolean?

Reply:
Once you return a boolean you can also move this code out of the else block.